src/app/dashboard/components/total/forecast.tsx

```diff
@@ -0,0 +1,181 @@
+import { EXPENSES_CATEGORIES, INCOMES_CATEGORIES } from "@/lib/categories";
+import { AppContext } from "@/lib/context";
+import {
+    formatDateToChartDate,
+    getFutureDatesAxisX,
+    getRangeAxisX,
+    parseDateToISO,
+    roundToTwoDecimal,
+    sortTransactions,
+    getDateRange, // Neue Funktion importiert
+} from "@/lib/utils";
+import React, { useContext, useEffect, useState } from "react";
+import {
+    XAxis,
+    YAxis,
+    CartesianGrid,
+    Tooltip,
+    ResponsiveContainer,
+    Legend,
+    LineChart,
+    Line,
+} from "recharts";
+import { DashboardNoDataCard } from "../ui/dashboard-no-data-card";
+import { DashboardCard } from "../ui/dashboard-card";
+import { SummaryTooltip } from "../ui/graph-utils";
+import { getTotalExpenses, getTotalIncomes } from "@/lib/calculations";
+
+interface ChartData {
+    name: string;
+    expense: number;
+    income: number;
+}
+
+function calculateFutureTrend(lastValue: number, futureDays: number, avg: number): number[] {
+    const futureValues: number[] = [];
+    futureValues.push(lastValue);
+
+    for (let i = 1; i < futureDays; i++) {
+        const nextValue = roundToTwoDecimal(futureValues[i - 1] + avg);
+        futureValues.push(nextValue);
+    }
+
+    return futureValues;
+}
+
+export default function SummaryChart() {
+    const { filteredTransactions, currency, selected } = useContext(AppContext);
+    const [data, setData] = useState<ChartData[]>([]);
+
+    useEffect(() => {
+        console.log("Selected value updated:", selected); // Log the selected value
```

Comment:
Please, remove all the unnecessary comments. For future occasions, I am not telling you not to rely on the AI to find more optimal solutions or improve your code, but accepting everything the AI gives you without even worrying about what it generates shows a real lack of concern